### PR TITLE
Improve documentation in Astrobee command dictionary

### PR DESCRIPTION
**Pull request by trey0**
I have always wanted the Astrobee command dictionary to be self-documenting to the point that a new guest scientist or FSW developer could get a clear idea of what each command is intended to be used for, and how it fits into the conops (for example, what are the steps of perching?). With these changes, it should be much closer to that point.

The caveat is that I'm not as close to the code as the FSW team, so some of my documentation might be wrong! In many places, it is based on either ancient memory or a kind of shallow reading of the code.

I recommend starting code review by opening doc/AstrobeeCommandDictionary.html in a browser.

In the schema: Mostly I just changed the "notes" field in many places, which should have no functional effect. In a few places, I also added a "choices" field, documented apparently available choices that weren't listed before, or specified a logical default value where none was present before. These changes propagate to output source files (e.g. cause some new macros to be defined), but again, it looks like they should have no functional effect.

I also updated the schema processing script slightly in the process of getting it to run on my MacBook.

### CODE REVIEW

**Review by kbrowne15 — CHANGES_REQUESTED** [review]

File: astrobee/commands/freeFlyerPlanSchema.json [hunk 1]
```diff
@@ -1127,7 +1168,7 @@
       "type": "CommandSpec",
       "id": "Admin.loadNodelet",
       "isRapidNative": false,
-      "notes": "Command used to load a nodelet in the system. Doesn't work with nodes running on the HLP.",
+      "notes": "This command is not implemented. The intent is that it will load a ROS nodelet in the Astrobee flight software.<p/>A note for developers: there is an internal service advertised by the system monitor that provides a method very similar to this command, but as of this writing, there does not appear to be any way to invoke it through a standard Astrobee API call. Also note, the underlying system monitor service can't load nodelets that run on the HLP.",
```
[the remark this answers is not in the document]
Comment (trey0) [reply astrobee/commands/freeFlyerPlanSchema.json]: Ok, fixed in https://github.com/trey0/astrobee/commit/ef356efee2945102367626be429e1f183fd033fd

File: astrobee/commands/freeFlyerPlanSchema.json [hunk 2]
```diff
@@ -1163,7 +1208,7 @@
       "type": "CommandSpec",
       "id": "Admin.unloadNodelet",
       "isRapidNative": false,
-      "notes": "Command used to unload a nodelet in the system. Doesn't work with nodes running on the HLP. With great power comes great responsibility! Don't unload a nodelet crucial to the system!!",
+      "notes": "For internal use only. Unloads a ROS nodelet in Astrobee's flight software. This command is defined only so as to make it available in the system monitor's fault response table. It is the usual response if initialization of a nodelet fails. There does not appear to be any way to invoke this command through a standard Astrobee API call. It doesn't work for nodes running on the HLP.",
```
[the remark this answers is not in the document]
Comment (trey0) [reply astrobee/commands/freeFlyerPlanSchema.json]: Ok, fixed in https://github.com/trey0/astrobee/commit/ef356efee2945102367626be429e1f183fd033fd

File: astrobee/commands/freeFlyerPlanSchema.json [hunk 3]
```diff
@@ -1092,6 +1130,7 @@
       "type": "CommandSpec",
       "id": "Admin.reacquirePosition",
       "isRapidNative": false,
+      "notes": "Switch Astrobee's localization to use the general-purpose pipeline.<p/>This command is equivalent to calling Admin.switchLocalization with the \"MappedLandmarks\" pipeline, which also has the side-effect of resetting the EKF, like calling Admin.resetEKF. It could be useful as a way to manually recover localization if Astrobee is lost because it is using the wrong localization pipeline (e.g., if docking or perching approach got interrupted, and targets are no longer in view).",
```
[the remark this answers is not in the document]
Comment (trey0) [reply astrobee/commands/freeFlyerPlanSchema.json]: Ok, fixed in https://github.com/trey0/astrobee/commit/ef356efee2945102367626be429e1f183fd033fd

File: astrobee/commands/freeFlyerPlanSchema.json [hunk 4]
```diff
@@ -964,7 +996,7 @@
       "type": "CommandSpec",
       "id": "Admin.shutdown",
       "isRapidNative": true,
-      "notes": "Put free flyer in hibernate power mode.",
+      "notes": "Puts Astrobee in hibernate power mode.<p/>An Astrobee that is hibernating can't receive Astrobee API commands. However, it can later be remotely awakened using Admin.wake, if it is on the dock and not physically switched off.",
```
[the remark this answers is not in the document]
Comment (trey0) [reply astrobee/commands/freeFlyerPlanSchema.json]: Ok, fixed in https://github.com/trey0/astrobee/commit/ef356efee2945102367626be429e1f183fd033fd

File: astrobee/commands/freeFlyerPlanSchema.json [hunk 5]
```diff
@@ -144,36 +150,47 @@
...
     }
   ],
   "commandSpecs": [
 
     {
       "type": "CommandSpec",
       "id": "Admin.wake",
-      "notes": "This command wakes astrobee from a hibernated state into a nominal state.",
-      "javaNotes": "Do not use this command. It is a dock command, not an astrobee command.",
+      "notes": "Wakes an Astrobee from a hibernating state (only EPS powered) into a ready state (main processors powered, flight software stack running, ready to receive Astrobee API commands). The Astrobee in question must be docked. Because each wake command is routed through a dock berth, which Astrobee to wake is specified using the berth number. See also Admin.wakeSafe.<p/>Note: For context about how to wake an Astrobee on the ISS, see the procedure IRG-FFTEST207a Astrobee Quick Wakeup and Checkout. This command is usually issued using a terminal menu configured in the astrobee_ops repo. After wake-up is commanded, additional network configuration may be needed before an Astrobee on the ISS can actually receive API commands from the ground.",
```
[the remark this answers is not in the document]
Comment (trey0) [reply astrobee/commands/freeFlyerPlanSchema.json]: Ok, fixed in https://github.com/trey0/astrobee/commit/ef356efee2945102367626be429e1f183fd033fd

File: astrobee/commands/freeFlyerPlanSchema.json [hunk 6]
```diff
@@ -833,8 +865,7 @@
       "type": "CommandSpec",
       "id": "Settings.setEnableAutoReturn",
       "isRapidNative": false,
-      "notes": "Command to allow auto return. Currently auto return is not implemented as intended. When this command is issued, astrobee will return to the dock and dock on berth 1. If there is already another astrobee on berth 1, the astrobees will collide and bad things will happen. DO NOT issue this command if berth 1 is occupied.",
-      "javaNotes": "Only issue this command if you are sure berth 1 is unoccupied.",
+      "notes": "For expert use only. Enables/disables auto-return to the dock. See Mobility.autoReturn.<p/>Note: Currently this flag appears to be ignored.",
```
[the remark this answers is not in the document]
Comment (trey0) [reply astrobee/commands/freeFlyerPlanSchema.json]: Ok, fixed in https://github.com/trey0/astrobee/commit/ef356efee2945102367626be429e1f183fd033fd

File: astrobee/commands/freeFlyerPlanSchema.json [hunk 7]
```diff
@@ -697,13 +723,15 @@
         {
           "type": "ParamSpec",
           "id": "telemetryName",
-          "parent": "Settings.TelemetryType"
+          "parent": "Settings.TelemetryType",
+          "notes": "The DDS telemetry topic to manage. Note: The \"CommStatus\" topic does not appear to actually be in use."
```
[the remark this answers is not in the document]
Comment (trey0) [reply astrobee/commands/freeFlyerPlanSchema.json]: Ok, fixed in https://github.com/trey0/astrobee/commit/ef356efee2945102367626be429e1f183fd033fd

File: astrobee/commands/freeFlyerPlanSchema.json [hunk 8]
```diff
@@ -193,34 +210,38 @@
...
+          "notes": "Target attitude"
         }
       ]
     },
 
     {
       "type": "CommandSpec",
       "id": "Settings.setCamera",
-      "notes": "Set camera parameters.",
+      "notes": "Set camera parameters. This command must be sent while the camera is not recording. Once parameters are set, use Settings.setCameraRecording to start recording.",
```
[the remark this answers is not in the document]
Comment (trey0) [reply astrobee/commands/freeFlyerPlanSchema.json]: Ok, fixed in https://github.com/trey0/astrobee/commit/ef356efee2945102367626be429e1f183fd033fd

File: astrobee/commands/freeFlyerPlanSchema.json [hunk 9]
```diff
@@ -230,35 +251,33 @@
...
-          "parent": "Settings.CameraName",
-          "notes": "Camera name"
+          "parent": "Settings.CameraName"
         },
         {
           "type": "ParamSpec",
           "id": "cameraMode",
-          "parent": "Settings.CameraMode",
-          "notes": "Desired camera mode. Either streaming or recording."
+          "parent": "Settings.CameraMode"
```
[the remark this answers is not in the document]
Comment (trey0) [reply astrobee/commands/freeFlyerPlanSchema.json:260]: Ok, fixed in https://github.com/trey0/astrobee/commit/ef356efee2945102367626be429e1f183fd033fd

File: astrobee/commands/freeFlyerPlanSchema.json [hunk 10]
```diff
@@ -267,7 +286,7 @@
     {
       "type": "CommandSpec",
       "id": "Settings.setCameraStreaming",
-      "notes": "Set streaming camera video to the ground.",
+      "notes": "Sets camera streaming live video to the ground on/off.",
```
[the remark this answers is not in the document]
Comment (trey0) [reply astrobee/commands/freeFlyerPlanSchema.json]: Ok, fixed in https://github.com/trey0/astrobee/commit/ef356efee2945102367626be429e1f183fd033fd

File: astrobee/commands/freeFlyerPlanSchema.json [hunk 11]
```diff
@@ -230,35 +251,33 @@
...
         {
           "type": "ParamSpec",
           "id": "frameRate",
           "valueType": "float",
           "minimum": 1,
-          "maximum": 60,
+          "maximum": 30,
           "unit": "Hz",
-          "notes": "Applies to both modes of camera."
+          "notes": "Applies to streaming and recording. Maximum frame rate varies by camera:<br/>- SciCam: 30 Hz <br/>- NavCam, DockCam: 15 Hz<br/>- HazCam, PerchCam: 5 Hz"
```
[the remark this answers is not in the document]
Comment (trey0) [reply astrobee/commands/freeFlyerPlanSchema.json]: Ok, fixed in https://github.com/trey0/astrobee/commit/ef356efee2945102367626be429e1f183fd033fd

File: astrobee/commands/freeFlyerPlanSchema.json [hunk 12]
```diff
@@ -182,7 +199,7 @@
       "type": "CommandSpec",
       "id": "Mobility.simpleMove6DOF",
       "isRapidNative": true,
-      "notes": "Astrobee teleop move command",
+      "notes": "Astrobee move command. Using the default trapezoidal planner, the move has the following steps:<br/>- If not in holonomic mode, the robot rotates to face the target position.<br/>- Once rotation is complete, the robot translates in a straight line to the target position.<br/>- Once translation is complete, the robot rotates to the target attitude.",
```
[the remark this answers is not in the document]
Comment (trey0) [reply astrobee/commands/freeFlyerPlanSchema.json]: Ok, fixed in https://github.com/trey0/astrobee/commit/8a80326356bb67ab919e23ab2f696670914e96a3

File: astrobee/commands/freeFlyerPlanSchema.json [hunk 13]
```diff
@@ -469,7 +485,7 @@
       "type": "CommandSpec",
       "id": "Plan.runPlan",
       "isRapidNative": false,
-      "notes": "Run the loaded plan",
+      "notes": "Runs the currently loaded plan. If the plan was previously paused with Plan.pausePlan, this command will resume execution where it previously left off. See plan.setPlan.",
```
[the remark this answers is not in the document]
Comment (trey0) [reply astrobee/commands/freeFlyerPlanSchema.json:489]: Ok, fixed in https://github.com/trey0/astrobee/commit/8a80326356bb67ab919e23ab2f696670914e96a3

File: astrobee/commands/freeFlyerPlanSchema.json [hunk 14]
```diff
@@ -943,7 +975,7 @@
       "type": "CommandSpec",
       "id": "Plan.wait",
       "isRapidNative": false,
-      "notes": "Pause plan for specified duration. Do nothing if docked/perched, otherwise station keep.",
+      "notes": "Temporarily pauses the running plan. Execution will resume automatically after the specified duration. See plan.setPlan.",
```
[the remark this answers is not in the document]
Comment (trey0) [reply astrobee/commands/freeFlyerPlanSchema.json]: Ok, fixed in https://github.com/trey0/astrobee/commit/8a80326356bb67ab919e23ab2f696670914e96a3

File: astrobee/commands/freeFlyerPlanSchema.json [hunk 15]
```diff
@@ -277,22 +296,21 @@
...
+          "notes": "Set to true/false to enable/disable streaming live video to the ground."
         }
       ]
     },
 
     {
       "type": "CommandSpec",
       "id": "Settings.setCameraRecording",
-      "notes": "Set camera to record video.",
+      "notes": "Set camera video onboard recording on/off.",
```
[the remark this answers is not in the document]
Comment (trey0) [reply astrobee/commands/freeFlyerPlanSchema.json]: Ok, fixed in https://github.com/trey0/astrobee/commit/8a80326356bb67ab919e23ab2f696670914e96a3

File: astrobee/commands/freeFlyerPlanSchema.json [hunk 16]
```diff
@@ -556,27 +572,31 @@
         {
           "type": "ParamSpec",
           "id": "name_",
-          "valueType": "string"
+          "valueType": "string",
+          "default": "",
+          "notes": "This field was originally intended to allow specifying what reference frame the center of mass is expressed in. However, in the current implementation, the reference frame is hard-coded to be the body frame, and this field is ignored."
```
[the remark this answers is not in the document]
Comment (trey0) [reply astrobee/commands/freeFlyerPlanSchema.json]: Ok, fixed in https://github.com/trey0/astrobee/commit/8a80326356bb67ab919e23ab2f696670914e96a3

**Review by kbrowne15 — APPROVED** [review]

**Pull request merged**